framework/include/userobject/NearestPointBase.h

```diff
@@ -79,51 +79,36 @@ class NearestPointBase : public ElementIntegralVariableUserObject
    * @param p The point.
    * @return The UserObject closest to p.
    */
-  UserObjectType * nearestUserObject(const Point & p) const;
+  std::shared_ptr<UserObjectType> nearestUserObject(const Point & p) const;
```

Comment:
We do need a discussion.

I will say that this is definitely an instance of needing shared pointers.  The pointers are built and held within this object and need to continue to be held within this object while other objects might also need to store pointers to those objects.

A unique pointer is definitely not the right thing to use here.